src/classes/PermanentFileSystem.ts

```diff
@@ -344,6 +344,26 @@ export class PermanentFileSystem {
     }
   }
 
+  public folderPathExistsInCache(itemPath: string): boolean {
+    return this.folderCache.has(itemPath);
```

Comment:
This should also check the parent folder's list of folders (similar to `archiverRecrodPathExistsInCache`)